src/jpgimage.cpp

```diff
@@ -136,12 +135,8 @@ namespace Exiv2 {
     // Todo: Generalised from JpegBase::locateIptcData without really understanding
     //       the format (in particular the header). So it remains to be confirmed
     //       if this also makes sense for psTag != Photoshop::iptc
-    int Photoshop::locateIrb(const byte*     pPsData,
-                             long            sizePsData,
-                             uint16_t        psTag,
-                             const byte**    record,
-                             uint32_t *const sizeHdr,
-                             uint32_t *const sizeData)
+    int Photoshop::locateIrb(const byte* pPsData, long sizePsData, uint16_t psTag, const byte** record,
+                             uint32_t& sizeHdr, uint32_t& sizeData)
     {
         assert(record);
         assert(sizeHdr);
```

Comment:
@neheb I assume the failures you are seeing are because you forgot to remove these `assert` statements after switching pointer to reference